src/cartservice/cartstore/RedisCartStore.cs

```diff
@@ -175,6 +181,8 @@ public async Task<Hipstershop.Cart> GetCartAsync(string userId)
         {
             Console.WriteLine($"GetCartAsync called with userId={userId}");
 
+            using var parent = CartActivity.ActivitySource.StartActivity("GetCart", ActivityKind.Server);
```

Comment:
We will want to standardize on span names if we want the UI to link externals with transactions. In the case of the Frontend and AdService, their span names are prefixed with `hipstershop.<serviceName>/<method>`

One example of this in action is the [`GetAds` externals call](https://staging-one.newrelic.com/launcher/nr1-core.explorer?pane=eyJuZXJkbGV0SWQiOiJvcGVuLWluc3RydW1lbnRhdGlvbi1leHBsb3Jlci5leHRlcm5hbHMiLCJlbnRpdHlJZCI6Ik1UQXpORGN4TlRoOFJWaFVmRk5GVWxaSlEwVjhNVEF6T1RNNE16ZzBORGcxTkRRd01UUTNOdyIsInRyYW5zYWN0aW9uTmFtZSI6ImhpcHN0ZXJzaG9wLkFkU2VydmljZS9HZXRBZHMifQ==&sidebars[0]=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5hY3Rpb25zIiwiZW50aXR5SWQiOiJNVEF6TkRjeE5UaDhSVmhVZkZORlVsWkpRMFY4TVRBek9UTTRNemcwTkRnMU5EUXdNVFEzTnciLCJzZWxlY3RlZE5lcmRsZXQiOnsibmVyZGxldElkIjoib3Blbi1pbnN0cnVtZW50YXRpb24tZXhwbG9yZXIuZXh0ZXJuYWxzIn19&platform[timeRange][duration]=1800000&platform[$isFallbackTimeRange]=true&platform[filters]=ImB0YWdzLmluc3RydW1lbnRhdGlvbi5wcm92aWRlcmAgPSAnb3BlbnRlbGVtZXRyeSci) which looks for spans with the name `hipstershop.AdService/GetAds` and can link to that "transaction"

Reply:
If it helps @elucus and maybe @alanwest this naming comes from the standard convention for RPC span names: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/rpc.md#span-name

Reply:
Yea makes sense to have these span names conform to the RPC spec. We can do that in the meantime, though ideally, we wouldn't be manually creating these spans. The CartService is a completely custom web service. I've had on my mind is to port this service to use a .NET framework for which there is auto-instrumentation. Then we'd get these spans and correct names for free.

Reply:
Though this brings up a question... our UI relies on people and library authors to strictly follow OpenTelemetry's semantic conventions for things to be linked up correctly? Seems like there are a lot of opportunities for this to break down. Let's chat more... maybe office hours.